File: Group Anagrams/Group_Anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String, List<String>> mpp = new HashMap<>();
        for (String s : strs) {
            char[] chars = s.toCharArray();
            Arrays.sort(chars);
            String sortedStr = new String(chars);
            mpp.putIfAbsent(sortedStr, new ArrayList<>());
            mpp.get(sortedStr).add(s);
        }
        return new ArrayList<>(mpp.values());
    }
}

File: Group Anagrams/Group_Anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        mpp = defaultdict(list)
        for s in strs:
            sorted_str = "".join(sorted(s))
            mpp[sorted_str].append(s)
        return list(mpp.values())
            